File: jja6312/2월 8일/H-Index.java
import java.util.Arrays;

class Solution {
    public int solution(int[] citations) {
        //특정 숫자 이상인 숫자가 특정 숫자만큼 있는지를 찾으면됨.

        Arrays.sort(citations);

        int maxH=0;
        for(int i=0; i<citations.length; i++){
            int h = Math.min(citations[i], citations.length-i);
            maxH = Math.max(maxH,h);
        }

        int answer = maxH;

        return answer;
    }
}

File: jeong-hyuk/2월8일/H-index.py
def solution(citations):
    citations.sort(reverse=True)  # 논문 인용 횟수를 내림차순으로 정렬
    
    h_index = 0
    for i, citation in enumerate(citations):
        # 현재 논문 인용 횟수가 현재 인덱스보다 크거나 같으면 h_index를 갱신
        if citation >= i + 1:
            h_index = i + 1
        else:
            break  # 현재 논문 인용 횟수가 현재 인덱스보다 작으면 반복문 종료
            
    return h_index